astropy/utils/data.py

```diff
@@ -1019,90 +1019,103 @@ def download_file(remote_url, cache=False, show_progress=True, timeout=None):
     else:
         url_key = remote_url
 
-    try:
-        if cache:
-            # We don't need to acquire the lock here, since we are only reading
-            with _open_shelve(urlmapfn, True) as url2hash:
-                if url_key in url2hash:
-                    return url2hash[url_key]
+    url_key_list = url_key.split(',')
```

Comment:
I meant something like:
```
if not iterable:
    treat_as_single_url
else:  # if iterable like a list
    for item in iterable:
        treat_item_as_single_url
```

Anyway, just a thought. If it does not make sense, you can ignore.